src/ucp/rndv/rndv.c

```diff
@@ -336,40 +317,66 @@ static void ucp_rndv_complete_rma_put_zcopy(ucp_request_t *sreq, int is_frag_put
 
         atp_req->send.ep = sreq->send.ep;
         atp_req->flags   = 0;
-        ucp_rndv_req_send_ack(atp_req, sreq, sreq->send.rndv.remote_req_id,
-                              status, UCP_AM_ID_RNDV_ATP, "send_atp");
+        ucp_rndv_req_send_ack(atp_req, sreq, sreq->send.req_id.remote, status,
+                              UCP_AM_ID_RNDV_ATP, "send_atp");
     }
 
     ucp_request_send_buffer_dereg(sreq);
     ucs_assert(sreq->send.state.dt.dt.contig.md_map == 0);
     ucp_request_complete_send(sreq, status);
 }
 
+UCS_PROFILE_FUNC_VOID(ucp_rndv_complete_frag_rma_put_zcopy, (fsreq),
+                      ucp_request_t *fsreq)
+{
+    ucp_request_t *sreq = fsreq->super_req;
+
+    sreq->send.state.dt.offset += fsreq->send.length;
+
+    /* delete fragments send request */
+    ucp_request_put(fsreq);
+
+    /* complete send request after put completions of all fragments */
+    if (sreq->send.state.dt.offset == sreq->send.length) {
+        ucp_send_request_del_id(sreq);
+        ucp_rndv_complete_rma_put_zcopy(sreq, 1);
+    }
+}
+
 static void ucp_rndv_recv_data_init(ucp_request_t *rreq, size_t size)
 {
     rreq->status         = UCS_OK;
     rreq->recv.remaining = size;
 }
 
+ucs_status_t ucp_rndv_send_rts(ucp_request_t *sreq, uct_pack_callback_t pack_cb,
+                               size_t rts_body_size)
+{
+    size_t max_rts_size = sizeof(ucp_tag_rndv_rts_hdr_t) +
```

Comment:
it is wrong, AM uses its own rndv header, need to pass header size together with `rts_body_size`

Reply:
fixed